.github/workflows/post-merge-deploy.yml

```diff
@@ -0,0 +1,48 @@
+name: Package for Deployment
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  deploy:
+    name: Package for build
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    environment: ${{ matrix.environment }}
+    strategy:
+      matrix:
+        environment: [development, build]
+    concurrency:
+      group: post-merge-${{ matrix.environment }}
+      cancel-in-progress: false
+    env:
+      AWS_REGION: eu-west-2
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Build SAM app
```

Comment:
This should be a separate job before the upload job so the app is only built once. The SAM cache can be used to get the artifact.
Example for test harness: https://github.com/govuk-one-login/ipv-cri-common-lambdas/blob/d10f763f9289a9ad9677561a771d609e4e98ed07/.github/workflows/package-test-resources.yml#L62C1-L68C25